### PR TITLE
MGMT-8796: clear msg about cidr range

### PR DESCRIPTION
**Pull request by slaviered**
CIDR block size should be large enough to accommodate all the hosts' IPs. When a user chooses a smaller range than required, the error message is not clear enough.

This PR adds a recommendation to the error message to guide the users.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/network/machine_network_cidr.go [hunk 1]
```diff
@@ -92,11 +93,36 @@ func VerifyVip(hosts []*models.Host, machineNetworkCidr string, vip string, vipN
...
+	if err != nil {
+		log.WithError(err).Errorf("can't parse machine cidr %s", machineNetworkCidr)
+		return true
+	}
+
+	networkPrefix, bits := cidr.Mask.Size()
+	numOfHosts := len(hosts)
+	if numOfHosts == 1 {
+		//allow /31 network prefix and down (the smaller the prefix the larger the range)
+		return networkPrefix <= 31
```
Comment (ori-amizur) [inline internal/network/machine_network_cidr.go:118]: IPv6 ?

Comment (slaviered) [reply internal/network/machine_network_cidr.go:118]: replaced with bits and the minimum network width constants

File: internal/network/machine_network_cidr.go [hunk 2]
```diff
@@ -92,11 +93,36 @@ func VerifyVip(hosts []*models.Host, machineNetworkCidr string, vip string, vipN
...
+		//to accommodates hosts and vips
+		if !isMachineNetworkCidrBigEnough(hosts, machineNetworkCidr, log) {
+			msg = fmt.Sprintf("%s. The machine network range is too small for the cluster. Please redefine the network.", msg)
+		}
+		return errors.New(msg)
 	}
 	return nil
 }
 
+func isMachineNetworkCidrBigEnough(hosts []*models.Host, machineNetworkCidr string, log logrus.FieldLogger) bool {
```
Comment (ori-amizur) [inline internal/network/machine_network_cidr.go:107]: I think this should be in the cidr_validations.go file

**Pull request merged**